tools/explorer/pkg/stellar/stellar.go

```diff
@@ -105,53 +139,108 @@ func (w *Wallet) CreateAccount() (keypair.Full, error) {
 		SourceAccount: &sourceAccount,
 		Operations:    []txnbuild.Operation{&createAccountOp},
 		Timebounds:    txnbuild.NewTimeout(300),
-		Network:       w.getNetworkPassPhrase(),
+		Network:       w.GetNetworkPassPhrase(),
 	}
 
 	txeBase64, err := tx.BuildSignEncode(w.keypair)
 	if err != nil {
-		return keypair.Full{}, errors.Wrap(err, "failed to get build transaction")
+		return errors.Wrap(err, "failed to get build transaction")
 	}
 
 	// Submit the transaction
 	_, err = client.SubmitTransactionXDR(txeBase64)
 	if err != nil {
 		hError := err.(*horizonclient.Error)
-		return keypair.Full{}, errors.Wrap(hError, "error submitting transaction")
+		return errors.Wrap(hError, "error submitting transaction")
 	}
+	return nil
+}
 
-	// Set the trustline
-	sourceAccount, err = w.getAccountDetails(newKp.Address())
+func (w *Wallet) setupTrustline(newKp *keypair.Full, sourceAccount hProtocol.Account, client *horizonclient.Client) error {
+	sourceAccount, err := w.GetAccountDetails(newKp.Address())
 	if err != nil {
-		return keypair.Full{}, errors.Wrap(err, "failed to get account details")
+		return errors.Wrap(err, "failed to get account details")
 	}
 	changeTrustOp := txnbuild.ChangeTrust{
 		SourceAccount: &sourceAccount,
 		Line: txnbuild.CreditAsset{
 			Code:   w.asset.String(),
-			Issuer: w.getIssuer(),
+			Issuer: w.GetIssuer(),
 		},
 	}
 	trustTx := txnbuild.Transaction{
 		SourceAccount: &sourceAccount,
 		Operations:    []txnbuild.Operation{&changeTrustOp},
 		Timebounds:    txnbuild.NewTimeout(300),
-		Network:       w.getNetworkPassPhrase(),
+		Network:       w.GetNetworkPassPhrase(),
 	}
 
-	txeBase64, err = trustTx.BuildSignEncode(newKp)
+	txeBase64, err := trustTx.BuildSignEncode(newKp)
 	if err != nil {
-		return keypair.Full{}, errors.Wrap(err, "failed to get build transaction")
+		return errors.Wrap(err, "failed to get build transaction")
 	}
 
 	// Submit the transaction
 	_, err = client.SubmitTransactionXDR(txeBase64)
 	if err != nil {
 		hError := err.(*horizonclient.Error)
-		return keypair.Full{}, errors.Wrap(hError.Problem, "error submitting transaction")
+		return errors.Wrap(hError.Problem, "error submitting transaction")
 	}
+	return nil
+}
 
-	return *newKp, nil
+func (w *Wallet) setupEscrowMultisig(newKp *keypair.Full, sourceAccount hProtocol.Account, client *horizonclient.Client) error {
+	if len(w.signers) >= 3 {
```

Comment:
I'd prefer it if this condition is inverted, i.e.:

```
if len(w.signers) < 3 {
    // not enough signers, don't add multisig
    return nil
}
```

This way it is immediately clear that this function will take no further action in this case, and that this is indeed a terminal condition

Reply:
@LeeSmet  What do you mean by `activeEscrowAccount` ?